flight/states/state.py

```diff
@@ -0,0 +1,75 @@
+"""Base state class for competition stages"""
+
+import logging
+from mavsdk import System
+from flight.state_settings import StateSettings
+
+
+class State:
+    """
+    Base flight state class
+
+    Attributes
+    ----------
+    state_settings : StateSettings
+        Parameters of flight plan, i.e. title description, etc.
+
+    Methods
+    -------
+    run(drone: System) -> None
+        Runs the code for each state and return the next, or None if at the end of the state machine
+    check_arm_or_arm(drone: System) -> None
+        Verifies if drone is armed, and if not, arms it
+    """
+
+    def __init__(self, state_settings: StateSettings) -> None:
+        """
+        Initializes base state with mission's Settings
+
+        Parameters
+        ----------
+        state_settings : StateSettings
+            Flight mission attributes & competition data
+        """
+        logging.info("State %s has begun", self.name)
+        self.state_settings: StateSettings = state_settings
+
+    async def run(self, drone: System) -> None:
+        """
+        Flight mission code for each state
+
+        Parameters
+        ----------
+        drone : System
+            MAVSDK drone object used to manipulate drone position & attitude
+        """
+        return
```

Comment:
Should raise an error here that says this is the superclass function and shouldn't be called.